core/cap-0046-02.md

```diff
@@ -124,72 +120,34 @@ The maximum WASM contract size will be introduced as a `ConfigSettingEntry`(see
 
 It is set during the protocol version upgrade using a new `ConfigSettingEntry`,
 with `configSettingID` == `CONFIG_SETTING_CONTRACT_MAX_SIZE_BYTES`, and
-`contractMaxSizeBytes` == `16384`. The valid values for
-`contractMaxSizeBytes` are[0, 256000] (inclusive).
-
-#### No contract-driven WASM installation
-
-This CAP does not specify a way to install WASM sources from within a
-contract. 
-
-This is done to encourage efficient code reuse and deduplication: if the
-contract was allowed to install the WASM code, then we'd need to store it
-twice (in the installer contract and in the source entry).
+`contractMaxSizeBytes` == `65536`. The valid values for
+`contractMaxSizeBytes` are [5000, UINT32_MAX] (inclusive).
 
 ### Instantiating contracts using `InvokeHostFunctionOp`
 
 Contracts can be instantiated via `InvokeHostFunctionOp` with
 `HOST_FUNCTION_TYPE_CREATE_CONTRACT` host function type.
 
 The function accepts `CreateContractArgs` struct that defines the input for
-building the contract identifier preimage (`contractID` field) and the contract
-source reference (`source` field).
-
-All the preimage types besides `ENVELOPE_TYPE_CONTRACT_ID_FROM_CONTRACT` can be
-built from the `contractID` field.
+building the contract identifier preimage (`contractIDPreimage` field) and the
```

Comment:
We also need to add a paragraph on authorization and the expected authorization payload. Authorization has to be performed by the deployer address. Payload pretty simple - just credentials + `CreateContractArgs`.

Reply:
Added a paragraph but realized it's not accurate. The credentials can still be of type address.

Reply:
Hmm I need to look into this a little more.

Reply:
Yeah, there is nothing special about the credentials. You can just say that the credentials in the entry have to match the addressed used to infer the contract id.

Reply:
Updated